nix-store.el

```diff
@@ -22,5 +28,152 @@ PATH the path within /nix/store to realise"
    :noquery t
    :name (format "*nix-store*<%s>" path)))
 
+(defvar-local nix-buffer-store-path nil "Buffer-local object holding an `nix-store-path` object")
+
+(defclass nix-store-path ()
+  ((path :initarg :path :accessor nix-store-path-path)
+    (status :initarg :status :initform 'realised)
+    (hash :initarg :hash)
+    (size :initarg :size)
+    (derivers :initarg :derivers)
+    (references :initarg :references)
+    (referrers :initarg :referrers)
+    (requisites :initarg :requisites))
+  "Nix-Store-Path Class that holds all information of the path that
+is displayed")
+
+(cl-defmethod nix-store-fill-data ((object nix-store-path))
+  (oset object :size (nix-store--query 'size (nix-store-path-path object)))
+  (oset object :hash (nix-store--query 'hash (nix-store-path-path object)))
+  (oset object :referrers (nix-store--query 'referrers (nix-store-path-path object)))
+  (oset object :requisites (nix-store--query 'requisites (nix-store-path-path object)))
+  (oset object :references (nix-store--query 'references (nix-store-path-path object)))
+  (oset object :derivers (nix-store--query 'deriver (nix-store-path-path object)))
+  object)
+
+(defun nix-store--query (argument path)
```

Comment:
The `--store` arg could be useful here. For instance, you can query stuff in cache.nixos.org if you pass --store cache.nixos.org:

```
nix-store --query --references /nix/store/w7i8xqvmjmdxgh9hn4h1gvqy1qn4rvz9-git-2.32.0 --store https://cache.nixos.org
```

I think this works on older Nix version.

Reply:
Good to know about the `--store` arg. However, since it only seems to work with older nix versions I want to refrain from introducing a new argument to the function. I would much rather support it in a form of a let binding to `nix-store-dir` like:

```elisp
(let ((nix-store-dir "https://cache.nixos.org"))
    ;;
  )
```
This would make it easier to write functions that just replace this variable ( buffer locally ) and then gain a view into another nix-store.